ord_schema/dataset_templating.py

```diff
@@ -0,0 +1,139 @@
+# Copyright 2020 Open Reaction Database Project Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Script and functions for enumerating a dataset using a template reaction and
+a spreadsheet file.
+
+The templating code has specific expectations for how the reaction pbtxt and
+spreadsheet are defined, namely that placeholder values in the pbtxt begin and
+end with a "$" (dollar sign) and that these match a unique column header in the
+spreadsheet file.
+
+Example usage:
+
+* For normal dataset generation from a template reaction:
+  $ python dataset_templating.py --input_template=my_reaction.pbtxt
```

Comment:
Naming nit, feel free to ignore: I find "input_" to be redundant in this context.